server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeAggregator.java

```diff
@@ -156,54 +167,164 @@ private void finishLeaf() {
         }
     }
 
+    /** Return true if the provided field has multiple values per document in the leaf **/
+    private boolean isMultiValued(LeafReaderContext context, SortField sortField) throws IOException {
```

Comment:
maybe call it `maybeMultivalued` or something like that since this might still return `true` on a single-valued field

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/48399/commits/03141a547a3011fa4fa49c60a58e59d85ec520e0